### PR TITLE
V3 support 

### PR DESCRIPTION
**Pull request by perryism**
# Purpose
Support voice base v3 api

# Implementation
1. add v3 client module (copy from v2....)
1. use the v3 config
1. change v3 response to adapt the new response structure

### CODE REVIEW

File: lib/voicebase/v3/client.rb [hunk 1]
```diff
@@ -0,0 +1,153 @@
+module VoiceBase
+  module V3
+    module Client
+      def self.extended(client, args = {})
+        client.api_host     = client.args[:host] || ENV.fetch('VOICEBASE_V3_API_HOST', 'https://apis.voicebase.com')
+        client.api_endpoint = client.args[:api_endpoint] || ENV.fetch('VOICEBASE_V3_API_ENDPOINT', '/v3')
```
Comment (frankywahl) [inline lib/voicebase/v3/client.rb]: Should this come from some configurable singleton or something?

File: lib/voicebase/v3/client.rb [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            uri + '/access/users/admin/tokens',
+            basic_auth: auth,
+            headers: {
+              'User-Agent'   => @user_agent,
+              'Accept'       => 'application/json'
+            }
+          ), api_version)
+        @token = VoiceBase::Client::Token.new(response.tokens.any? && response.tokens.first.fetch("token"))
+      rescue NoMethodError => ex
+        raise VoiceBase::AuthenticationError, response.status_message
```
Comment (frankywahl) [inline lib/voicebase/v3/client.rb]: are we sure we have a `response`? should we then put the `response` outside the `rescue` block. 

Maybe we don't even need a `rescue`, if between the `response` and `@token` lines, we `raise VoiceBase::AuthenticationError if something?(response)`

File: lib/voicebase/v3/client.rb [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            headers: r.headers,
+            body: r.body
+        )
+        VoiceBase::Response.new(response, api_version)
+      end
+
+      # I presume this method exists for parity with the V1 API however we are not using it
+      def get_media(args = {}, headers = {})
+        raise ArgumentError, "Missing argument :media_id" unless args[:media_id]
+        url = if args[:media_id]
```
Comment (frankywahl) [inline lib/voicebase/v3/client.rb]: why is there an `if` statement here given it was just checked above?

File: lib/voicebase/v3/client.rb [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+          uri + "/media/#{args[:media_id]}"
+        elsif args[:external_id]
+          uri + "/media?externalID=#{args[:external_id]}"
+        else
+          raise ArgumentError, "Missing argument :media_url or :media_file"
```
Comment (frankywahl) [inline lib/voicebase/v3/client.rb]: what does `media_url` have to do do here? I don't see any checks for it.

File: lib/voicebase/v3/client.rb [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        end
+        if args[:external_id]
```
Comment (frankywahl) [inline lib/voicebase/v3/client.rb]: can't we just make this a `required` parameter to the method?

File: lib/voicebase/v3/client.rb (hunk 1 above)
Comment (perryism) [reply lib/voicebase/v3/client.rb]: I am just following the patterns from v1 and v2. 
https://github.com/usertesting/voicebase-client-ruby/blob/master/lib/voicebase/v1/client.rb#L11-L12
https://github.com/usertesting/voicebase-client-ruby/blob/master/lib/voicebase/v2/client.rb#L8-L9

File: lib/voicebase/v3/client.rb (hunk 2 above)
Comment (perryism) [reply lib/voicebase/v3/client.rb]: I actually didn't implement this. it's from v2.  https://github.com/usertesting/voicebase-client-ruby/blob/master/lib/voicebase/v2/client.rb#L13-L25
Maybe this should be removed and let V3 should extend V2.

File: lib/voicebase/v3/client.rb (hunk 3 above)
Comment (perryism) [reply lib/voicebase/v3/client.rb]: I also didn't touch this. this is from v2. https://github.com/usertesting/voicebase-client-ruby/blob/master/lib/voicebase/v2/client.rb#L43-L59

Comment (perryism) [reply lib/voicebase/v3/client.rb]: maybe the new commit would make more sense https://github.com/usertesting/voicebase-client-ruby/pull/16/commits/28bfe248708eef26141fc76c659031438f35e8dc

File: .travis.yml [hunk 6]
```diff
@@ -1,6 +1,5 @@
 language: ruby
 rvm:
-  - 2.0
```
Comment (frankywahl) [inline .travis.yml]: maybe also add 2.6 since that is what orders is running of off?

File: lib/voicebase/v3/response.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+        ok?
+      end
+
+      def media_id
+        voicebase_response['mediaId']
+      end
+
+      def transcript_ready?
+        voicebase_response['status'].casecmp(TRANSCRIPT_READY_STATUS) == 0
```
Comment (frankywahl) [inline lib/voicebase/v3/response.rb]:
```suggestion
        voicebase_response['status']&.downcase == TRANSCRIPT_READY_STATUS
```

**Pull request merged**